dist/pythonlibs/riotnode/riotnode/node.py

```diff
@@ -0,0 +1,205 @@
+                         encoding=encoding, codec_errors=codec_errors,
+                         **kwargs)
+
+    def expect(self, pattern, *args, **kwargs):
+        # pylint:disable=arguments-differ
+        try:
+            return super().expect(pattern, *args, **kwargs)
+        except (pexpect.TIMEOUT, pexpect.EOF) as exc:
+            raise self._pexpect_exception(exc, pattern)
+
+    def expect_exact(self, pattern, *args, **kwargs):
+        # pylint:disable=arguments-differ
+        try:
+            return super().expect_exact(pattern, *args, **kwargs)
+        except (pexpect.TIMEOUT, pexpect.EOF) as exc:
+            raise self._pexpect_exception(exc, pattern)
+
+    @staticmethod
+    def _pexpect_exception(exc, pattern):
+        """Tweak pexpect exception.
+
+        * Put the calling 'pattern' as value
+        * Remove exception context
+        """
+        exc.pexpect_value = exc.value
+        exc.value = pattern
+
+        # Remove exception context
+        exc.__cause__ = None
+        exc.__traceback__ = None
+        return exc
+
+
+class RIOTNode():
+    """Class abstracting a RIOTNode in an application.
+
+    This should abstract the build system integration.
+
+    :param application_directory: relative directory to the application.
+    :param env: dictionary of environment variables that should be used.
+                These overwrites values coming from `os.environ` and can help
+                define factories where environment comes from a file or if the
+                script is not executed from the build system context.
+
+    Environment variable configuration
+
+    :environment BOARD: current RIOT board type.
+    :environment RIOT_TERM_START_DELAY: delay before `make term` is said to be
+                                        ready after calling.
+    """
+
+    TERM_SPAWN_CLASS = TermSpawn
+    TERM_STARTED_DELAY = int(os.environ.get('RIOT_TERM_START_DELAY') or 3)
+
+    MAKE_ARGS = ()
+    RESET_TARGETS = ('reset',)
+
+    def __init__(self, application_directory='.', env=None):
+        self._application_directory = application_directory
+
+        # TODO I am not satisfied by this, but would require changing all the
+        # environment handling, just put a note until I can fix it.
+        # I still want to show a PR before this
+        # I would prefer getting either no environment == os.environ or the
+        # full environment to use.
+        # It should also change the `TERM_STARTED_DELAY` thing.
+        self.env = os.environ.copy()
+        self.env.update(env or {})
+
+        self.term = None  # type: pexpect.spawn
+
+        self.logger = logging.getLogger(__name__)
+
+    @property
+    def application_directory(self):
+        """Absolute path to the current directory."""
+        return os.path.abspath(self._application_directory)
+
+    def board(self):
+        """Return board type."""
+        return self.env['BOARD']
```

Comment:
```suggestion
        return self.env.gety('BOARD')
```
so it doesn't return a key error, or is this on purpose?